CONTRIBUTING.md

```diff
@@ -1,41 +1,78 @@
+When contributing to eXist-db you should branch one of the development branches above, your branch should be named in one of two ways:
+
+* `feature/<name-of-my-feature>`
+    This naming convention should be used when contributing new features to eXist-db. For example `feature/xquery31-sliding-window`
+* `hotfix/<name-of-my-fix>`
+  This naming convention should be used when contributing bug fixes to eXist-db. For example `feature/xquery31-sliding-window`
 
-Pull Requests are reviewed and tested before they're merged by the core development team.
-However, we have one golden rule, even within the core team: **never merge your own pull request**. This simple-but-important rule ensures that at least two people have considered the change.
+Additionally, if you are back-porting a feature or bug fix to a previous version of eXist-db, you should prefix your branch name with a `V.x.x/` where `V` is the major version number, for example: `6.x.x/feature/xquery31-sliding-window`.
+
+## Code Formatting
+All new Java code is expected to be formatted inline with the [IntelliJ Default Style for Java code](https://www.jetbrains.com/help/idea/configuring-code-style.html#configure-code-style-schemes). 
+
+
+## Commit Naming
+Commits to eXist-db by developers *must* follow a naming convention. This convention expects a prefix to be prepended to each of your commit messages. The prefixes enable us to better understand the commit and also to generate Change Logs. The prefixes in use are:  
+
+* `[feature]`
+    This should be used when a commit adds a new feature.
+* `[bugfix]`
+    This should be used when a commit addresses a bug or issue.
+* `[refactor]`
+    This should be used when a commit is simply refactoring existing code.
+* `[ignore]`
+    This should be used when code is cleaned up by automated means, e.g. reformatting.
+* `[doc]`
+    This should be used for documentation.
+* `[test]`
+    This should be used when a commit solely contains changes to existing tests or adds further tests.
+* `[ci]`
+    This should be used when a commit solely makes changes to CI configuration.
+
+In addition any commit that addresses a GitHub issues, should have an additonal line in its commit message like:
+```
+Closes https://github.com/eXist-db/exist/issues/<github-issue-number>
+```
+
+For example, here is a correctly formatted commit message:
+
+```
+[bugfix] Store the path of Jar files in each EXPath Package's `.exist/classpath.txt` file relative to the package's `content/` sub-folder.
+Closes https://github.com/eXist-db/exist/issues/4901
+```
+
+## Pull Requests and Code Review
+Pull Requests are reviewed and tested before they're merged by the eXist-db Core Development Team.
+We have a policy around how Pull Requests are reviewed in a timely and fair manner. That policy is available here - [Community Code Review and Merge Policy for the exist-db/exist Git Repository](PR-CODE-REVIEW-POLICY.md). 
```

Comment:
s/Pull Requests/pull requests

Reply:
Thanks @joewiz I pushed a commit to address this.